src/uv-common.c

```diff
@@ -513,7 +513,18 @@ int uv_fs_scandir_next(uv_fs_t* req, uv_dirent_t* ent) {
   uv__dirent_t** dents;
   uv__dirent_t* dent;
 
-  unsigned int* nbufs = uv__get_nbufs(req);
+  unsigned int* nbufs;
+  
+  /* Did we fail in the request? */
+  if (req->result < 0)
+    return req->result;
+  
+  /* Did we find any files? */
+  if (!req->ptr)
+    return UV_EOF;
```

Comment:
Is this case not covered [here](https://github.com/muflub/libuv/blob/3dfb862f8a79df367c1689421aa6e998f5fcdf78/test/test-fs.c#L2172)?